geode-assembly/src/distributedTest/java/org/apache/geode/rest/internal/web/controllers/RestAPICompatibilityTest.java

```diff
@@ -0,0 +1,196 @@
+    MemberVM locator2 =
+        cluster.startLocatorVM(1, 0, oldVersion,
+            x -> x.withConnectionToLocator(locatorPort1).withHttpService());
+    int locatorPort2 = locator2.getPort();
+    cluster
+        .startServerVM(2, oldVersion, s -> s.withRegion(RegionShortcut.PARTITION, "region")
+            .withConnectionToLocator(locatorPort1, locatorPort2));
+    cluster
+        .startServerVM(3, oldVersion, s -> s.withRegion(RegionShortcut.PARTITION, "region")
+            .withConnectionToLocator(locatorPort1, locatorPort2));
+
+    // Roll locators to the current version
+    cluster.stop(0);
+    // gradle sets a property telling us where the build is located
+    final String buildDir = System.getProperty("geode.build.dir", System.getProperty("user.dir"));
+    locator1 = cluster.startLocatorVM(0, l -> l.withHttpService().withPort(locatorPort1)
+        .withConnectionToLocator(locatorPort2)
+        .withSystemProperty("geode.build.dir", buildDir));
+    cluster.stop(1);
+
+    cluster.startLocatorVM(1,
+        x -> x.withConnectionToLocator(locatorPort1).withHttpService().withPort(locatorPort2)
+            .withConnectionToLocator(locatorPort1)
+            .withSystemProperty("geode.build.dir", buildDir));
+
+    gfsh.connectAndVerify(locator1);
+    gfsh.execute("list members");
+    // Execute REST api calls to from the new locators to the old servers to ensure that backwards
+    // compatibility is maintained
+
+    executeAndValidatePOSTRESTCalls(locator1.getHttpPort());
+    executeAndValidateGETRESTCalls(locator1.getHttpPort());
+
+  }
+
+  void executeAndValidatePOSTRESTCalls(int locator) throws Exception {
+
+    CloseableHttpClient httpClient = HttpClients.createDefault();
+    for (Map.Entry<String, String[]> entry : postRESTAPICalls.entrySet()) {
+      // Skip the test is the version is before the REST api was introduced.
+      if (TestVersion.compare(oldVersion, entry.getValue()[2]) < 0) {
+        continue;
+      }
+      HttpPost post =
+          new HttpPost("http://localhost:" + locator + entry.getKey());
+      post.addHeader("Content-Type", "application/json");
+      post.addHeader("Accept", "application/json");
+      StringEntity jsonStringEntity =
+          new StringEntity(entry.getValue()[0], ContentType.DEFAULT_TEXT);
+      post.setEntity(jsonStringEntity);
+      CloseableHttpResponse response = httpClient.execute(post);
+
+      HttpEntity entity = response.getEntity();
+      InputStream content = entity.getContent();
+
+      BufferedReader reader = new BufferedReader(new InputStreamReader(content));
+      String line;
+      StringBuilder sb = new StringBuilder();
+      while ((line = reader.readLine()) != null) {
+        sb.append(line);
+      }
+      JsonNode jsonObject = mapper.readTree(sb.toString());
+      String statusCode = jsonObject.findValue("statusCode").textValue();
+      assertTrue(statusCode.equals("ACCEPTED") || statusCode.contains("OK"));
```

Comment:
I strongly recommend AssertJ rather than JUnit Assert especially in this case. The AssertJ failure message has much better information about why it failed.

The JUnit assertTrue will fail with a message like this:
```
java.lang.AssertionError 
        at org.junit.Assert.fail(Assert.java:87) 
        at org.junit.Assert.assertTrue(Assert.java:42) 
        at org.junit.Assert.assertTrue(Assert.java:53)
	at org.apache.geode.rest.internal.web.controllers.RestAPICompatibilityTest.executeAndValidatePOSTRESTCalls(RestAPICompatibilityTest.java:168)
	at org.apache.geode.rest.internal.web.controllers.RestAPICompatibilityTest.restCommandExecutedOnLatestLocatorShouldBeBackwardsCompatible (RestAPICompatibilityTest.java:135)
```
If you use AssertJ:
```
assertThat(statusCode).satisfiesAnyOf(
  value -> assertThat(value).isEqualTo("ACCEPTED"),
  value -> assertThat(value).contains("OK")
);
```
Then you will get a failure message like this:
```
org.assertj.core.error.MultipleAssertionsError: 
The following 2 assertions failed:
1) 
Expected :"[ACCEPTED]"
Actual   :"[FAIL]"

	at org.assertj.core.error.AssertionErrorCreator.multipleAssertionsError(AssertionErrorCreator.java:106)
	at org.assertj.core.api.AbstractAssert.multipleAssertionsError(AbstractAssert.java:947)
	at org.assertj.core.api.AbstractAssert.satisfiesAnyOfAssertionsGroups(AbstractAssert.java:942)
	at org.assertj.core.api.AbstractAssert.satisfiesAnyOf(AbstractAssert.java:855)
	at org.apache.geode.AwaitTest.executeAndValidatePOSTRESTCalls(AwaitTest.java:50)
	at org.apache.geode.rest.internal.web.controllers.RestAPICompatibilityTest.executeAndValidatePOSTRESTCalls(RestAPICompatibilityTest.java:168)
	at org.apache.geode.rest.internal.web.controllers.RestAPICompatibilityTest.restCommandExecutedOnLatestLocatorShouldBeBackwardsCompatible (RestAPICompatibilityTest.java:135)
```